src/components/LocationsList.js

```diff
@@ -69,32 +71,31 @@ class LocationsList extends InjectedComponent<InjectedProps, Props> {
       id: item.id,
     });
 
-  _renderItem = ({ item }: { item: Row<Location> }): React.Node => {
-    if (item.value.isLoading()) {
-      return <LoadingListItem />;
-    }
-
-    if (item.value.hasError()) {
-      return <ErrorListItem />;
-    }
-
-    const location = item.value.getValue();
+  _renderRow = ({ info, ...swipeableStateProps }): React.Node => (
+    <SwipeableLoaderRow
+      {...swipeableStateProps}
+      loader={info.item.value}
```

Comment:
Hmm... shouldn't this be the `LoadObject`? It doesn't make sense to pass the value and call it loader.

Reply:
info.item = Row<TEntity>, Row = { key: string, value: LoadObject<TEntity> }, so value here is LoadObject, I should probably just rename value to loader in Row... I'm just trying to not deviate from core FlatList interface too much, that why sometimes there are  naming collisions such this :/